dashboard/modules/job/job_head.py

```diff
@@ -347,15 +426,23 @@ async def tail_job_logs(self, req: Request) -> Response:
                 status=aiohttp.web.HTTPBadRequest.status_code,
             )
 
+        driver_agent_http_address = job.driver_agent_http_address
+        driver_node_id = job.driver_node_id
+        if driver_node_id not in self._agents_pool:
+            self._agents_pool = JobAgentSubmissionClient(driver_agent_http_address)
+        job_agent_client = self._agents_pool[driver_node_id]
```

Comment:
Is the agent_http_address the address of the agent that started the job, or the agent on the node where the job driver is running?  (These two can be different, right?)

Also, I thought any agent could get the logs from any job.  Is it a performance optimization to use this particular agent here?

Reply:
`agent_http_address` is the node address where the job driver is running, `SupervisorActor` will put it into the internal_vk after launching the driver process.
no, an agent can only get the logs of which job driver running on the same node.